packages/components/src/components/DistributionChart.tsx

```diff
@@ -23,16 +24,58 @@ export type DistributionPlottingSettings = {
   showControls: boolean;
 } & DistributionChartSpecOptions;
 
+export type Plot = {
+  distributions: Distribution[];
+};
+
 export type DistributionChartProps = {
-  distribution: Distribution;
+  plot: Plot;
   width?: number;
   height: number;
   actions?: boolean;
 } & DistributionPlottingSettings;
 
+export function defaultPlot(distribution: Distribution): Plot {
+  return { distributions: [distribution] };
+}
+export function makePlot(expression: {
+  [key: string]: squiggleExpression;
+}): Plot | void {
+  if (expression["distributions"].tag === "array") {
+    let distributions: Distribution[] = expression["distributions"].value
+      .map((x) => {
+        if (x.tag === "distribution") {
+          return x.value;
+        }
+      })
+      .filter((x): x is Distribution => x !== undefined);
+    return { distributions };
```

Comment:
I'm a bit worried about this: if a user doesn't know about this feature and decides to return a record with `distributions` key then all other keys will be silently ignored.

I understand that plot objects as first-class citizens in core language would be harder to implement (but we should do that eventually anyway, I guess). But maybe `makePlot` should check that (1) `distributions` is the only key in the expression; don't skip over non-distribution elements but fallback to the default rendering?

Reply:
I agree, it would be better to fail faster